### PR TITLE
feat(integrations): Add Jira Server to the Integrations Page

### PR DESCRIPTION
**Pull request by lauryndbrown**
Add Jira Server as an integration on the integration page. The build integration methods are all mocked out. To be fixed and added at a later date.

### CODE REVIEW

**Comment by lauryndbrown** [comment]
So far got the integration to show up. Will be starting on building out the provider soon
screen shot 2018-11-13 at 12 59 27 pm

File: src/sentry/integrations/jira_server/integration.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    can_add = False
+
+    def get_pipeline_views(self):
+        return []
+
+    def build_integration(self, state):
+        return {
+            'provider': 'jira_server',
+            'external_id': state['clientKey'],
+            'name': 'Jira Server',  # TODO(lb): definitely wrong
```
Comment (markstory) [inline src/sentry/integrations/jira_server/integration.py]: Its wrong, but its enough to get us started :smile:

**Comment by lauryndbrown** [comment]
@markstory the `Learn more` button gives me a blank page. Something on the js side is not happy. I think it has to do with jira server not having an image? Not totally sure. It's getting nothing back from render. 

Other than that I think the code does it's job.

File: tests/sentry/integrations/jira_server/test_integration.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+from __future__ import absolute_import
+
+from sentry.integrations.jira_server import JiraServerIntegrationProvider
+from sentry.identity.jira_server import JiraServerIdentityProvider
+from sentry.testutils import IntegrationTestCase
+
+
+class JiraServerIntegrationTest(IntegrationTestCase):
```
[the remark this answers is not in the document]
Comment (lauryndbrown) [reply tests/sentry/integrations/jira_server/test_integration.py]: `raven.js?9565:381 Uncaught TypeError: Cannot read property 'avatar' of undefined
    at Object.onOrganizationStoreUpdate (organizationBadge.jsx?b74f:70)
    at EventEmitter.eventHandler (PublisherMethods.js?0b0d:44)
    at EventEmitter.emit (index.js?22dc:137)
    at Store.trigger (PublisherMethods.js?0b0d:63)
    at Store.load (organizationsStore.jsx?2f9a:62)
    at Object.componentWillUnmount (app.jsx?1971:136)
    at Object.target.componentWillUnmount (method_decorator.js?123a:54)
    at Object.chainedFunction (factory.js?0b03:615)
    at Object.chainedFunction [as componentWillUnmount] (factory.js?0b03:614)
    at callComponentWillUnmountWithTimer (react-dom.development.js?1a2e:14601)
    at HTMLUnknownElement.callCallback (react-dom.development.js?1a2e:145)
    at HTMLUnknownElement.wrapped (raven.js?9565:377)
    at Object.invokeGuardedCallbackDev (react-dom.development.js?1a2e:195)
    at invokeGuardedCallback (react-dom.development.js?1a2e:248)
    at safelyCallComponentWillUnmount (react-dom.development.js?1a2e:14608)
    at commitUnmount (react-dom.development.js?1a2e:14840)
    at unmountHostComponents (react-dom.development.js?1a2e:15149)
    at commitDeletion (react-dom.development.js?1a2e:15180)
    at commitAllHostEffects (react-dom.development.js?1a2e:15851)
    at HTMLUnknownElement.callCallback (react-dom.development.js?1a2e:145)
    at HTMLUnknownElement.wrapped (raven.js?9565:377)
    at Object.invokeGuardedCallbackDev (react-dom.development.js?1a2e:195)
    at invokeGuardedCallback (react-dom.development.js?1a2e:248)
    at commitRoot (react-dom.development.js?1a2e:16039)
    at completeRoot (react-dom.development.js?1a2e:17462)
    at performWorkOnRoot (react-dom.development.js?1a2e:17390)
    at performWork (react-dom.development.js?1a2e:17294)
    at performSyncWork (react-dom.development.js?1a2e:17266)
    at requestWork (react-dom.development.js?1a2e:17154)
    at scheduleWork (react-dom.development.js?1a2e:16948)
    at Object.enqueueSetState (react-dom.development.js?1a2e:11622)
    at Object.Component.setState (react.development.js?0a94:413)
    at Object.eval (connect.js?3125:23)
    at EventEmitter.eventHandler (PublisherMethods.js?0b0d:44)
    at EventEmitter.emit (index.js?22dc:116)
    at Store.trigger (PublisherMethods.js?0b0d:63)
    at Store.onOpenModal (modalStore.jsx?bd43:26)
    at EventEmitter.eventHandler (PublisherMethods.js?0b0d:44)
    at EventEmitter.emit (index.js?22dc:116)
    at Function.trigger (PublisherMethods.js?0b0d:63)
    at eval (PublisherMethods.js?0b0d:74)
    at wrapped (raven.js?9565:377)`

**Comment by lauryndbrown** [comment]
> @markstory the `Learn more` button gives me a blank page. Something on the js side is not happy. I think it has to do with jira server not having an image? Not totally sure. It's getting nothing back from render.
> 
> Other than that I think the code does it's job.

```
raven.js?9565:381 Uncaught TypeError: Cannot read property 'avatar' of undefined
    at Object.onOrganizationStoreUpdate (organizationBadge.jsx?b74f:70)
    at EventEmitter.eventHandler (PublisherMethods.js?0b0d:44)
    at EventEmitter.emit (index.js?22dc:137)
    at Store.trigger (PublisherMethods.js?0b0d:63)
    at Store.load (organizationsStore.jsx?2f9a:62)
    at Object.componentWillUnmount (app.jsx?1971:136)
    at Object.target.componentWillUnmount (method_decorator.js?123a:54)
    at Object.chainedFunction (factory.js?0b03:615)
    at Object.chainedFunction [as componentWillUnmount] (factory.js?0b03:614)
    at callComponentWillUnmountWithTimer (react-dom.development.js?1a2e:14601)
    at HTMLUnknownElement.callCallback (react-dom.development.js?1a2e:145)
    at HTMLUnknownElement.wrapped (raven.js?9565:377)
    at Object.invokeGuardedCallbackDev (react-dom.development.js?1a2e:195)
    at invokeGuardedCallback (react-dom.development.js?1a2e:248)
    at safelyCallComponentWillUnmount (react-dom.development.js?1a2e:14608)
    at commitUnmount (react-dom.development.js?1a2e:14840)
    at unmountHostComponents (react-dom.development.js?1a2e:15149)
    at commitDeletion (react-dom.development.js?1a2e:15180)
    at commitAllHostEffects (react-dom.development.js?1a2e:15851)
    at HTMLUnknownElement.callCallback (react-dom.development.js?1a2e:145)
    at HTMLUnknownElement.wrapped (raven.js?9565:377)
    at Object.invokeGuardedCallbackDev (react-dom.development.js?1a2e:195)
    at invokeGuardedCallback (react-dom.development.js?1a2e:248)
    at commitRoot (react-dom.development.js?1a2e:16039)
    at completeRoot (react-dom.development.js?1a2e:17462)
    at performWorkOnRoot (react-dom.development.js?1a2e:17390)
    at performWork (react-dom.development.js?1a2e:17294)
    at performSyncWork (react-dom.development.js?1a2e:17266)
    at requestWork (react-dom.development.js?1a2e:17154)
    at scheduleWork (react-dom.development.js?1a2e:16948)
    at Object.enqueueSetState (react-dom.development.js?1a2e:11622)
    at Object.Component.setState (react.development.js?0a94:413)
    at Object.eval (connect.js?3125:23)
    at EventEmitter.eventHandler (PublisherMethods.js?0b0d:44)
    at EventEmitter.emit (index.js?22dc:116)
    at Store.trigger (PublisherMethods.js?0b0d:63)
    at Store.onOpenModal (modalStore.jsx?bd43:26)
    at EventEmitter.eventHandler (PublisherMethods.js?0b0d:44)
    at EventEmitter.emit (index.js?22dc:116)
    at Function.trigger (PublisherMethods.js?0b0d:63)
    at eval (PublisherMethods.js?0b0d:74)
    at wrapped (raven.js?9565:377)```

**Comment by markstory** [comment]
The broken 'Learn More' link sounds like something we can fix when we get a proper logo and integration description.

**Review by markstory — APPROVED** [review]

**Pull request merged**